### PR TITLE
feat: add metamask pending page

### LINKED ISSUE #32: Pending page while we are not connected to metamask
image

This is how the page looks like. I think it would be better to hide part of the page or show a something that tells the user we wait for metamask to connect.

Let me know if not doable

### PR DESCRIPTION
**Pull request by ChenRen93**
Closes #32

### CODE REVIEW

File: src/app.tsx [hunk 1]
```diff
@@ -1,15 +1,26 @@
...
+    const accounts = await window.ethereum.request({ method: "eth_requestAccounts" });
+    accounts.length > 0 ? setMetamaskConnected(true) : setMetamaskConnected(false);
+  };
+
   if (!window.ethereum) {
     return <MetaMaskError />;
   }
 
-  return <MainPage />;
+  return metamaskConnected ? <MainPage /> : <MetaMaskPending />;
```
Comment (Nebulis) [inline src/app.tsx:23]: great. If I click on cancel on metamask, the pending page stays.

Comment (ChenRen93) [reply src/app.tsx:27]: done. I added a check for code 4001 and bring users to the main page when they cancel the metamask.

Comment (Nebulis) [reply src/app.tsx:27]: image

Should only display the connect button

Comment (ChenRen93) [reply src/app.tsx:27]: Done.

File: src/components/header.tsx [hunk 2]
```diff
@@ -16,9 +16,11 @@ interface walletInfoType {
   walletBalance?: string;
 }
 
-export const Header: FunctionComponent = () => {
-  const [isConnected, setIsConnected] = useState(false);
-
+interface headerProps {
```
Comment (Nebulis) [inline src/components/header.tsx:19]: HeaderProps

File: src/page/main-page.tsx [hunk 3]
```diff
@@ -50,76 +50,83 @@ export const MainPage: FunctionComponent = () => {
...
-              >
-                {blockData.text}
-              </a>
-            );
-          })}
-      </div>
-
-      {documentStoreStatus && activeFeatureBlock === "issue-revoke" && activeSubFeatureBlock === "revoke" && (
-        <RevokeBlock documentStoreAddress={documentStoreAddress} />
+      <Header isConnected={isConnected} setIsConnected={setIsConnected} />
```
Comment (Nebulis) [inline src/page/main-page.tsx:57]: This is weird, why the header handles the state of connection?

Comment (ChenRen93) [reply src/page/main-page.tsx:57]: the header contains the Metamask connect button and action on retrieving wallet informations. so i use the state to check if the wallet is connected before displaying the rest of the content

File: src/app.tsx [hunk 4]
```diff
@@ -1,15 +1,30 @@
...
+  useEffect(() => {
+    checkMetaMaskAccount();
+  });
+
+  const checkMetaMaskAccount = async () => {
+    try {
+      const accounts = await window.ethereum.request({ method: "eth_requestAccounts" });
+      accounts.length > 0 ? setMetamaskConnected(true) : setMetamaskConnected(false);
+    } catch (e) {
+      e.code === 4001 ? setMetamaskConnected(true) : console.debug(e);
```
Comment (Nebulis) [inline src/app.tsx:19]: is this documented somewhere ? would be interesting to link to.

Comment (ChenRen93) [reply src/app.tsx:19]: https://docs.metamask.io/guide/ethereum-provider.html#using-the-provider.
 I read the documentation from here and did some modifications.
 i will add a comment at the top to display the link

Comment (Nebulis) [reply src/app.tsx:21]: https://docs.metamask.io/guide/ethereum-provider.html#errors

it's documented here. can add the link into the code ?

Comment (ChenRen93) [reply src/app.tsx:22]: done.

**Review by Nebulis — APPROVED** [review]

**Pull request merged**